crates/bevy_ecs/src/schedule/set.rs

```diff
@@ -118,11 +120,14 @@ impl<T> SystemSet for SystemTypeSet<T> {
 /// [`Schedule::add_systems`](super::Schedule::add_systems) or
 /// [`Schedule::configure_sets`](super::Schedule::configure_sets).
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
-pub struct AnonymousSet(usize);
+pub struct AnonymousSet(u64);
 
 impl AnonymousSet {
-    pub(crate) fn new(id: usize) -> Self {
-        Self(id)
+    /// New unique anonymous set.
+    pub fn new() -> Self {
+        // Assuming it never overflows.
```

Comment:
Let's not assume here, and validate that this never overflows. This constructor is in no hot path that I'm aware of, and can easily handle the extra overflow check.

Reply:
> This constructor is in no hot path

It is cheap even if it is on hot path, because integer comparison and never taken unlikely branch are virtually free.

If a machine does a billion atomic increment per second (more than current hardware can do), it would take 500 years to overflow, so I assumed it never overflows.

Anyway, I'll add overflow check.